### PR TITLE
MT-14-Topic 5 home work

### PR DESCRIPTION
**Pull request by AliakseiDziardziyashenka**
Created new folder for hw in solution explorer ,added 6 new projects

### CODE REVIEW

**Comment by batonkarton** [comment]
Greate folder structure - well done!

File: MentoringHomeWork/HW5/GoodDay/Greeting.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+namespace GoodDay;
+
+public class Greeting
+{
+    public void Main()
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/GoodDay/Greeting.cs:5]: Let's talk a bit more about naming - you choose a convention for your HW - like a class for an HW with the method "Main" for activity - not bad but we can improve it to make it more reasonable.

File: MentoringHomeWork/HW5/HW.05.Task2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using HW._05.Task2;
+
+Pyramid task = new Pyramid();
+task.Main();
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task2/Program.cs:6]: image
It looks like it's not working locally on my machine - we will discuss it.

File: MentoringHomeWork/HW5/GoodDay/Greeting.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            Console.WriteLine("Good day");
+        }
+        else if (date1.Hours >= 15 && date1.Hours < 23)
+        {
+            Console.WriteLine("Good evening");
+        }
+        else
+        {
+            Console.WriteLine("Good night");
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/GoodDay/Greeting.cs]: Clean code - well done!

File: MentoringHomeWork/HW5/HW.05.Task4/Counts.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+            {
+                count++;
+            }
+            else if (item == 'А')
+            {
+                count++;
+            }
+        }
+
+        Console.WriteLine(count);
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task4/Counts.cs:24]: Clean code - well done!

**Review by batonkarton — CHANGES_REQUESTED** [review]
Much better - well done!

File: MentoringHomeWork/HW5/HW.05.Task1/Program.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using HW._05.Task1;
+
+CalculeteCount calc = new CalculeteCount();
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task1/Program.cs]: CalculAte

File: MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
+using System.Runtime.InteropServices;
+
+namespace HW._05.Task1;
+
+public class CalculeteCount
+{
+    public void Main()
+    {
+        Console.WriteLine("Please, enter your number");
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs]: "Please, enter your number:" - minor

File: MentoAD.sln [hunk 7]
```diff
@@ -15,6 +15,30 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Task4", "MentoringHomeWork\
...
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "LattersCount", "MentoringHomeWork\HW5\LattersCount\LattersCount.csproj", "{83EF544B-7064-46CC-AFB4-82D7FDD35AB3}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HW.05.Task2", "MentoringHomeWork\HW5\HW.05.Task2\HW.05.Task2.csproj", "{70EB9210-8462-4551-A70C-86712BA222F2}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HW.05.Task3", "MentoringHomeWork\HW5\HW.05.Task3\HW.05.Task3.csproj", "{B4A144F5-73AF-457D-A60B-9FC92E34EBEB}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HW.05.Task4", "MentoringHomeWork\HW5\HW.05.Task4\HW.05.Task4.csproj", "{6E9CF6C8-28E1-4D1A-9D37-1320B0269B3D}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Home Work 5", "Home Work 5", "{00726010-3262-4CE1-9F31-A18E2F921EF5}"
```
Comment (batonkarton) [inline MentoAD.sln]: Please let's use pattern - "HomeWork5" - it;s more Dev style and can be efficient in CI/Cd in the future

File: MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+using System.Runtime.InteropServices;
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs]: Removed unused libraries - clean code

File: MentoringHomeWork/HW5/HW.05.Task3/TimeOfDay.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+public class TimeOfDay
+{
+    public void Main()
+    {
+        Console.WriteLine("Time now " + DateTime.Now.TimeOfDay);
+
+        TimeSpan date1 = DateTime.Now.TimeOfDay;
+
+        switch (date1.Hours)
+        {
```
Comment (batonkarton) [inline MentoringHomeWork/HW5/HW.05.Task3/TimeOfDay.cs:12]:
```
            case >= 9 and < 12:
                Console.WriteLine("Good morning");
                break;
            case >= 12 and < 15:
                Console.WriteLine("Good day");
                break;
            case >= 15 and < 23:
                Console.WriteLine("Good evening");
                break;
            default:
                Console.WriteLine("Good night");
                break;
```
just Rider refactoring - let's try

File: MentoringHomeWork/HW5/HW.05.Task1/Program.cs (hunk 5 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/HW.05.Task1/Program.cs]: resolved

File: MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs (hunk 6 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs]: resolved

File: MentoAD.sln (hunk 7 above)
Comment (AliakseiDziardziyashenka) [reply MentoAD.sln]: resolved

File: MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs (hunk 8 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/HW.05.Task1/CalculeteCount.cs]: resolved

File: MentoringHomeWork/HW5/HW.05.Task3/TimeOfDay.cs (hunk 9 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/HW.05.Task3/TimeOfDay.cs:12]: resolved

File: MentoringHomeWork/HW5/GoodDay/Greeting.cs (hunk 1 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/GoodDay/Greeting.cs:5]: Next time its been better

File: MentoringHomeWork/HW5/HW.05.Task2/Program.cs (hunk 2 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW5/HW.05.Task2/Program.cs:6]: resolved

**Review by batonkarton — APPROVED** [review]

**Pull request merged**